### PR TITLE
lightning: reduce the "unable to get keyspace name" log level to DEBUG

### LINKED ISSUE #54232: Lightning wants CONFIG privilege to query keyspace name and it confuses on-prem users
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

Run Lightning in logical mode with default configuration, especially leaving `[tikv-importer] keyspace-name` empty. 

Do not use the `root` user to restore to the downstream TiDB. Rather, create a new user and grant `SELECT,INSERT,UPDATE,DELETE,CREATE,DROP,ALTER` on `*.*` to that user ([as described in the doc](https://docs.pingcap.com/tidb/stable/tidb-lightning-requirements)).

### 2. What did you expect to see? (Required)

Everything works normally.

### 3. What did you see instead (Required)

There is a warning saying the CONFIG privilege is needed.

```
[2024/06/22 06:41:00.296 +00:00] [WARN] [lightning.go:565] ["unable to get keyspace name, lightning will use empty keyspace name"] [error="Error 1227 (42000): Access denied; you need (at least one of) the CONFIG privilege(s) for this operation"]
```

### 4. What is your TiDB version? (Required)

Lightning v7.5.1, v7.5.2, v8.1.0

### PR DESCRIPTION
**Pull request by kennytm**
### What problem does this PR solve?

Issue Number: close #54232

Problem Summary: When the keyspace-name is empty (default), and the user connecting to downstream TiDB is not a superuser, there will be a confusing "access denied" warning requesting CONFIG privilege. This confuses users to think that this privilege is actually required.

### What changed and how does it work?

Just reduce the log level to DEBUG.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by lance6716 — APPROVED** [review]

File: lightning/pkg/server/lightning.go [hunk 1]
```diff
@@ -567,7 +567,7 @@ func (l *Lightning) run(taskCtx context.Context, taskCfg *config.Config, o *opti
 		if keyspaceName == "" {
 			keyspaceName, err = getKeyspaceName(db)
 			if err != nil {
-				o.logger.Warn("unable to get keyspace name, lightning will use empty keyspace name", zap.Error(err))
+				o.logger.Debug("unable to get keyspace name, lightning will use empty keyspace name", zap.Error(err))
```
Comment (D3Hunter) [inline lightning/pkg/server/lightning.go:570]:
```suggestion
				o.logger.Info("unable to get keyspace name, lightning will use empty keyspace name", zap.Error(err))
```
maybe change to info, so we can know what's going on.

**Review by GMHDBJD — APPROVED** [review]
LGTM

File: lightning/pkg/server/lightning.go (hunk 1 above)
Comment (lance6716) [reply lightning/pkg/server/lightning.go:570]: there's another log in line 573

**Comment by lance6716** [comment]
/hold

wait to address comment

File: lightning/pkg/server/lightning.go (hunk 1 above)
Comment (kennytm) [reply lightning/pkg/server/lightning.go:570]: The user will still see the "access denied" message from `zap.Error(err)` though

**Comment by kennytm** [comment]
PTAL @D3Hunter

**Review by D3Hunter — APPROVED** [review]

**Pull request merged**